File: test/test2/src/main/java/Test8.java
public class Test8 {
    public static void main(String args[]) {
        int flag = 0;
        for (int i = 901; i < 1000; i++) {
            int flag1 = loop(i);
            if (flag1 > flag) {
                flag = flag1;
            }
        }
        System.out.print(flag);
    }

    private static int loop(int n) {
        int j;
        for (j = 1; n != 1; j++) {
            if (n % 2 == 1) {
                n = 3 * n + 1;
            } else {
                n = n / 2;
            }
        }
        return j;
    }
}

File: test/test3/test8.py
for i in range(100):
    j = i
    a = [1] * i
    k = 0
    flag = 0
    while j > 1:
        if k > i - 1:
            k = 0
        if a[k] == 0:
            k += 1
            continue
        else:
            flag += 1
            if flag == 2:
                a[k] = 0
                j -= 1
                flag = 0
            k += 1
    for m in range(i):
        if a[m] == 1:
            print m + 1

